### PR TITLE
API take one

### PR DESCRIPTION
**Pull request by cableman**
Extra! extra!... Read all about it ...

### CODE REVIEW

**Review by turegjorup — APPROVED** [review]
## Questions

This does not need to be addressed in this PR but should be considered going forward:
* Error handling/response for all "link" endpoints? How do we handle calls with non-existing id's?
* Do we use `::class`for entity references? E.g, `Slide::class` or `App\Entity\Slide` in `return $this->playlistRepository->getPaginator('App\Entity\Slide', $playListUlidObj, $page, $itemsPerPage);`
)
* Consider using `assert...instanceof` instead of `@var` e.g `assert($media instanceof Media);` This help the IDE in the same way with the added benefit of a run time error if the assumption is ever wrong.

File: config/api_platform/layout.yaml [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+          responses:
+            '200':
+              description: OK
+              content:
+                application/ld+json:
+                  examples:
+                    4-way-split:
+                      value:
+                        "@context": {
+                          "@vocab": "http://displayapiservice.local.itkdev.dk/docs.jsonld#",
```
[the remark this answers is not in the document]
Comment (cableman) [reply config/api_platform/layout.yaml:33]: Removed as we have agreed on not having examples in the yaml at this point in the development

File: config/api_platform/layout.yaml [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                          - "@context": "/contexts/ScreenLayoutRegions"
+                            "@id": "/v1/screen-layout-regions/01FFS94CHYW9268WFP1JF6H1VC"
+                            "@type": "ScreenLayoutRegions"
+                            title: Region 4
+                            gridArea:
+                              - d
+                    Complicated:
+                      value:
+                        "@context": {
+                          "@vocab": "http://displayapiservice.local.itkdev.dk/docs.jsonld#",
```
[the remark this answers is not in the document]
Comment (cableman) [reply config/api_platform/layout.yaml:73]: Removed as we have agreed on not having examples in the yaml at this point in the development

File: config/api_platform/layout.yaml [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+                            "@id": "/v1/screen-layout-regions/01FFS94CHYW9268WFP1JF6H1VC"
+                            "@type": "ScreenLayoutRegions"
+                            title: Region 6
+                            gridArea:
+                              - f
+                              - rr
+                    3x3:
+                      value:
+                        "@context": {
+                          "@vocab": "http://displayapiservice.local.itkdev.dk/docs.jsonld#",
```
[the remark this answers is not in the document]
Comment (cableman) [reply config/api_platform/layout.yaml:131]: Removed as we have agreed on not having examples in the yaml at this point in the development

File: config/api_platform/screen_groups.yaml [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+          responses:
+            '200':
+              description: OK
+              content:
+                application/ld+json:
+                  examples:
+                    Group:
+                      value:
+                        "@context": {
+                          "@vocab": "http://displayapiservice.local.itkdev.dk/docs.jsonld#",
```
[the remark this answers is not in the document]
Comment (cableman) [reply config/api_platform/screen_groups.yaml:35]: Removed as we have agreed on not having examples in the yaml at this point in the development

File: config/services.yaml [hunk 5]
```diff
@@ -11,6 +11,10 @@ services:
         autowire: true      # Automatically injects dependencies in your services.
         autoconfigure: true # Automatically registers your services as commands, event subscribers, etc.
 
+
+        bind:
+            $bindDefaultDataFormat: '%env(string:APP_DEFAULT_DATE_FORMAT)%'
```
[the remark this answers is not in the document]
Comment (cableman) [reply config/services.yaml:16]: Fixed

File: src/Controller/PlaylistScreenGetController.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        if (!Ulid::isValid($id)) {
+            throw new InvalidArgumentException();
+        }
+
+        $page = (int) $request->query->get('page', '1');
+        $itemsPerPage = (int) $request->query->get('itemsPerPage', '10');
+
+        $playListUlidObj = Ulid::fromString($id);
+
+        return $this->playlistRepository->getPaginator('App\Entity\Screen', $playListUlidObj, $page, $itemsPerPage);
```
[the remark this answers is not in the document]
Comment (cableman) [reply src/Controller/PlaylistScreenGetController.php:32]: Change to using `::class` I did not known that `Screen::class` would return the whole string with namespace :-)

File: src/Controller/PlaylistScreenRegionDeleteController.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        if (!(Ulid::isValid($id) && Ulid::isValid($regionId) && Ulid::isValid($playlistId))) {
+            throw new InvalidArgumentException();
+        }
+
+        $screenUlid = Ulid::fromString($id);
+        $regionUlid = Ulid::fromString($regionId);
+        $playlistUlid = Ulid::fromString($playlistId);
+
+        $this->playlistScreenRegionRepository->unlinkPlaylistsByScreenRegion($screenUlid, $regionUlid, $playlistUlid);
```
[the remark this answers is not in the document]
Comment (cableman) [reply src/Controller/PlaylistScreenRegionDeleteController.php:31]: They are tested inside the `unlinkPlaylistsByScreenRegion` as that is a database lookup and all database stuff is placed in the repository functions.

In this case an exception with "relation not found" with 404 will be thrown

File: src/Controller/PlaylistSlideGetController.php [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        if (!Ulid::isValid($id)) {
+            throw new InvalidArgumentException();
+        }
+
+        $page = (int) $request->query->get('page', '1');
+        $itemsPerPage = (int) $request->query->get('itemsPerPage', '10');
+
+        $slideUlidObj = Ulid::fromString($id);
+
+        return $this->slideRepository->getPaginator('App\Entity\PLaylist', $slideUlidObj, $page, $itemsPerPage);
```
[the remark this answers is not in the document]
Comment (cableman) [reply src/Controller/PlaylistSlideGetController.php:32]: Change to using `::class`

File: src/Controller/SlidePlaylistGetController.php [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        if (!Ulid::isValid($id)) {
+            throw new InvalidArgumentException();
+        }
+
+        $page = (int) $request->query->get('page', '1');
+        $itemsPerPage = (int) $request->query->get('itemsPerPage', '10');
+
+        $playListUlidObj = Ulid::fromString($id);
+
+        return $this->playlistRepository->getPaginator('App\Entity\Slide', $playListUlidObj, $page, $itemsPerPage);
```
[the remark this answers is not in the document]
Comment (cableman) [reply src/Controller/SlidePlaylistGetController.php:32]: Change to using ::class

**Comment by cableman** [comment]
@turegjorup Change the requested changes (with the exception of assert)

**Pull request merged**